### DISCUSSION TITLE
API explorer does not display filtering options

### DISCUSSION THREAD

**Discussion by xpusostomos** [specify/specify7#5932] [Feature Requests]
https://sp7demofish.specifycloud.org/documentation/api/tables/

It seems to me that maybe the online rest interface isn't setup correctly: 

GET
/api/specify/determiner/
Query multiple records from the Determiner table
"Filterring is supported by providing field values as GET parameters"

When I press "Try it out", there appears to be nowhere to enter field values. Assuming this is an oversight, perhaps the oversight occurs elsewhere too.

**To Reproduce**
Go to the determiner table. GET /api/specify/determiner/
Press "Try it out"
There is nowhere to enter values to filter on

**Expected behavior**
I presume the try it out interface should provide a way to enter field values for filtering.

**Comment by maxpatiiuk** [comment]
Hi

While filtering is supported by the API, it is not implemented in the visual API explorer as there are infinitely many possible variations on what filter values you could provide.

Filtering by fields looks like this:
```yaml
# Simple conditions:
/api/specify/determiner/?text3=test&text1_startswith=abc&number1_gte=1
# Query partial date
/api/specify/determiner/?determinedDate_year=2030
# Filter by a field in related table
/api/specify/determiner/?createdByAgent__lastname_icontains=J&determiner__type_contains=Organization,Group
```

Behind the scenes, we proxy all query parameters to Django's ORM. Thus, you can take a look at [Django documentation](https://docs.djangoproject.com/en/4.0/topics/db/queries/) to see a full list of available options. More specifically, take a look at [this Python method](https://github.com/specify/specify7/blob/aa434e5d24f9b88b3a2c2d892d8619f5d35e815e/specifyweb/specify/api.py#L782-L812) where we parse the filter parameters

Plus, the API explorer page already is feeling slow due to how many endpoints it has to display - if each endpoint gains additional hundreds of parameters, the page would become unusable.

Though, we should probably document this better - at least, have a link in the API explorer to a page that describes this in more details.
@grantfitzsimmons do you know who on the team would be interested in doing technical documentation for this? or else I could write a Github wiki page for that
